packages/core/admin/admin/src/content-manager/components/DynamicTable/CellContent/utils/tests/hasContent.js

```diff
@@ -0,0 +1,48 @@
+import hasContent from '../hasContent';
+
+describe('hasContent', () => {
+  it('returns true for text content', () => {
+    const normalizedContent = hasContent('text', 'content');
+    expect(normalizedContent).toEqual(true);
+  });
+
+  it('returns false for empty text content', () => {
+    const normalizedContent = hasContent('text', '');
+    expect(normalizedContent).toEqual(true);
+  });
+
+  it('returns false for undefined text content', () => {
+    const normalizedContent = hasContent('text', undefined);
+    expect(normalizedContent).toEqual(true);
+  });
+
+  it('extracts content from single components with content', () => {
+    const normalizedContent = hasContent(
+      'text',
+      { name: 'content' },
+      { mainField: 'name' },
+      { repeatable: false }
+    );
+    expect(normalizedContent).toEqual(true);
+  });
+
+  it('extracts content from single components without content', () => {
+    const normalizedContent = hasContent('text', {}, { mainField: 'name' }, { repeatable: false });
+    expect(normalizedContent).toEqual(false);
+  });
+
+  it('extracts content from repeatable components with content', () => {
+    const normalizedContent = hasContent(
+      'text',
+      [{ name: 'content' }, { name: 'content_2' }],
+      { mainField: 'name' },
```

Comment:
With the comment below the layout should be `{mainField: { type, name: 'name } }`